homeassistant/components/deconz/sensor.py

```diff
@@ -67,171 +70,163 @@
-            if isinstance(device, LightLevel) and isinstance(device.light_level, int)
-            else None,
-            update_key="lightlevel",
-            device_class=SensorDeviceClass.ILLUMINANCE,
-            state_class=SensorStateClass.MEASUREMENT,
-            native_unit_of_measurement=LIGHT_LUX,
-        )
-    ],
-    Power: [
-        DeconzSensorDescription(
-            key="power",
-            value_fn=lambda device: device.power if isinstance(device, Power) else None,
-            update_key="power",
-            device_class=SensorDeviceClass.POWER,
-            state_class=SensorStateClass.MEASUREMENT,
-            native_unit_of_measurement=POWER_WATT,
-        )
-    ],
-    Pressure: [
-        DeconzSensorDescription(
-            key="pressure",
-            value_fn=lambda device: device.pressure
-            if isinstance(device, Pressure)
-            else None,
-            update_key="pressure",
-            device_class=SensorDeviceClass.PRESSURE,
-            state_class=SensorStateClass.MEASUREMENT,
-            native_unit_of_measurement=PRESSURE_HPA,
-        )
-    ],
-    Temperature: [
-        DeconzSensorDescription(
-            key="temperature",
-            value_fn=lambda device: device.scaled_temperature
-            if isinstance(device, Temperature) and isinstance(device.temperature, int)
-            else None,
-            update_key="temperature",
-            device_class=SensorDeviceClass.TEMPERATURE,
-            state_class=SensorStateClass.MEASUREMENT,
-            native_unit_of_measurement=TEMP_CELSIUS,
-        )
-    ],
-    Time: [
-        DeconzSensorDescription(
-            key="last_set",
-            value_fn=lambda device: device.last_set
-            if isinstance(device, Time)
-            else None,
-            update_key="lastset",
-            device_class=SensorDeviceClass.TIMESTAMP,
-            state_class=SensorStateClass.TOTAL_INCREASING,
-        )
-    ],
-}
-
-
-COMMON_SENSOR_DESCRIPTIONS = [
-    DeconzSensorDescription(
+    common: bool = False
+    name_suffix: str = ""
+    old_unique_id_suffix: str = ""
+
+
+ENTITY_DESCRIPTIONS: tuple[DeconzSensorDescription, ...] = (
+    DeconzSensorDescription[AirQuality](
+        key="air_quality",
+        isinstance_fn=lambda device: isinstance(device, AirQuality),
+        value_fn=lambda device: device.air_quality,
+        update_key="airquality",
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    DeconzSensorDescription[AirQuality](
+        key="air_quality_ppb",
+        isinstance_fn=lambda device: isinstance(device, AirQuality),
+        value_fn=lambda device: device.air_quality_ppb,
+        update_key="airqualityppb",
+        name_suffix="PPB",
+        old_unique_id_suffix="ppb",
+        device_class=SensorDeviceClass.AQI,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=CONCENTRATION_PARTS_PER_BILLION,
```

Comment:
The AQI device class sensor should not have a unit of measurement.

https://developers.home-assistant.io/docs/core/entity/sensor#available-device-classes

Reply:
Thanks Martin! Resolved in https://github.com/home-assistant/core/pull/79380